### PR TITLE
DataGrid full-screen mode

### PR DESCRIPTION
**Pull request by andrzejewsky**
closes: #2412 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants

CONTAINERS=1

### CODE REVIEW

File: src/components/Datagrid/FullScreenContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    ...(isOpen ? openStyles : closedStyles),
+  };
+};
+
+interface FullScreenContainerProps {
+  open?: boolean;
+  className?: string;
+}
+
+const Portal = ({ className, children, open }) => {
```
Comment (andrzejewsky) [inline src/components/Datagrid/FullScreenContainer.tsx:61]: this file implements actually a modal with mounting/unmounting, will leave a comment here to make it more generic so we can apply it across all of our modals

File: src/components/Datagrid/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <CardTitle
+      title={title}
+      toolbar={<div className={classes.btnContainer}>{children}</div>}
+    />
+  );
+};
+
+Header.ButtonFullScreen = ButtonFullScreen;
```
Comment (krzysztofzuraw) [inline src/components/Datagrid/Header.tsx:76]: Nice pattern 👏🏻

File: src/components/Datagrid/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import React from "react";
+
+import CardTitle from "../CardTitle";
+
+const useStyles = makeStyles(
+  theme => ({
+    btnContainer: {
+      display: "flex",
+      flexDirection: "row-reverse",
+      gap: "10px",
```
Comment (Droniu) [inline src/components/Datagrid/Header.tsx:13]: theme.spacing(1)?

File: src/components/Datagrid/usePortal.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { useEffect } from "react";
+
+/**
+ * Glide automatically refers to the div with an id "portal"
+ * This hook allows to add additional classes for it
+ */
+const portalRoot = document.getElementById("portal");
+
+export const usePortal = ({ className }: { className: string }) => {
```
Comment (Droniu) [inline src/components/Datagrid/usePortal.ts:9]: If it's used just for styling can we name usePortalClasses or something like this

**Review by Droniu — APPROVED** [review]

**Review by Droniu — DISMISSED** [review]

**Review by Droniu — APPROVED** [review]

**Pull request merged**